imodqgis/widgets/unique_color_widget.py

```diff
@@ -107,6 +115,37 @@ def classify(self) -> None:
                 Qt.ItemIsEnabled | Qt.ItemIsEditable | Qt.ItemIsSelectable
             )
 
+    def _get_tiled_normalized_midpoints(self, n_elements, tilesize):
+        """
+        Create array with length n_class of rolling normalized midpoint values
+        which can be used to to fetch values on unique colors colormap.
+
+        Example
+        -------
+        >>> self._get_tiled_normalized_midpoints(n_elements=6, tilesize=4)
+        array([0.125, 0.375, 0.625, 0.875, 0.125, 0.375)]
+        """
+        return ((np.arange(n_elements) % tilesize) + 0.5) / tilesize
+
+    def get_colors_from_ramp_button(self) -> List[QColor]:
+        uniques = pd.Series(self.data).dropna().unique()
+        n_class = uniques.size
+        ramp = self.color_ramp_button.colorRamp()
+        n_colors = ramp.count()
+        # Cap n_colors at max 12, as for example some gradient colormaps have
+        # 255 stops. Too high values for n_colors create too subtle color
+        # shifts.
+        n_colors = np.min([n_colors, 12])
```

Comment:
12 seems like a weird number. I'm doubtful whether this the most robust approach.

Reply:
I changed the behaviour based on this feedback, if ramp type is "colorbrewer" or "random", it will take the cyclic normalized point values, else ("gradient", "cpt-city") it will use ``np.linspace(0.0, 1.0, n_class)``.